app.rb

```diff
@@ -0,0 +1,126 @@
+require_relative 'lib/database_connection'
+require_relative 'lib/item_repo'
+require_relative 'lib/order_repo'
+require_relative 'lib/item'
+
+class Application
+  def initialize(db_name, io, item_repo, order_repo)
+    DatabaseConnection.connect(db_name)
+    @io = io
+    @item_repo = item_repo
+    @order_repo = order_repo
+  end
+
+  def all_items
+    @io.puts "\nHere's a list of all shop items:\n\n"
+    items = @item_repo.all.sort_by { |item| item.id.to_i }
+    items.each do |item|
+      @io.puts "##{item.id} - #{item.name} - Unit price: #{item.unit_price} - Quantity: #{item.qty}"
+    end
+  end
+
+  def ask_respond(text, text_2 = nil)
+    @io.puts text
+    @io.puts text_2 if text_2 != nil
+    @io.gets.chomp
+  end
+
+  def proceed(item)
+    str = "\n Create item: <#{item.name} - #{item.unit_price} - #{item.qty}>? [Y/n]"
+    proceed = ask_respond(str)
+    @item_repo.create(item) if proceed.downcase == "y"
+  end
+
+  def new_item
+    item = Item.new
+    item.name = ask_respond("\nEnter item name:")
+    item.unit_price = ask_respond("\nEnter unit price:")
+    item.qty = ask_respond("\nEnter stock quantity:")
+    proceed(item)
+  end
+  
+  def all_orders
+    ord = "\nHere's a list of all orders:\n\n"
+    orders = @order_repo.all.sort_by { |order| order.id.to_i }
+    orders.each do |order|
+      ord += "  ##{order.id} - Customer: #{order.customer_name} - Placed: #{order.date_placed}\n"
+      full_order = @order_repo.order_with_items(order.id)
+      full_order.items.each do |item|
+        ord += "    * #{item.name} - Unit price: #{item.unit_price} - qty: #{item.qty}\n"
+      end
+    end
+    @io.puts ord
+  end
+
+  def make_item
+    user = ask_respond("\nEnter <item name>, <qty> to add to order", "Type 'Y' when done")
+    return false if user.downcase == "y"
+
+    # ***problem here if user enters nothing!***
+    new_item = Item.new
+    new_item.name = user.split(",")[0].strip
+    new_item.qty = user.split(",")[1].strip
```

Comment:
does this always return the right word? and not 'enter' as if its split at the comma then u end of with arrays of strings?

Reply:
it's a terrible piece of code that splits the response from the ```ask_respond``` method. Trouble is - if the user enters nothing then ask_respond will come back with nothing and lines 61/62 will fall over.